fem/fe.hpp

```diff
@@ -1396,24 +1501,29 @@ class Lagrange1DFiniteElement : public NodalFiniteElement
    mutable Vector rxxk;
 #endif
 public:
+   /// Construct the FiniteElement with the provided @a degree
    Lagrange1DFiniteElement (int degree);
    virtual void CalcShape(const IntegrationPoint &ip, Vector &shape) const;
    virtual void CalcDShape(const IntegrationPoint &ip,
                            DenseMatrix &dshape) const;
 };
 
+
 class P1TetNonConfFiniteElement : public NodalFiniteElement
```

Comment:
Missing class description (and I'm curious to find out what this is...)

Reply:
Honestly I don't know what this is either. It is a tet with nodes in odd places.  I think it might have had something to do with non conforming AMR on tets at one point, but I can't find anything using it at this point.  @tzanio @v-dobrev  any ideas on this one?

Reply:
This is the 3D tet equivalent of the Crouzeix-Raviart triangle element -- these are non-conforming H1 elements. For the tet element, the dofs are the centers of the 4 faces. For the triangle element, the dofs are the centers of the 3 edges.